test/batchReceiver.spec.ts

```diff
@@ -421,4 +423,533 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.defer();
+
+    await testPeekMsgsLength(queueClient, 1);
+
+    if (!sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    deferredMsgs = await queueClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.complete();
+
+    await testPeekMsgsLength(queueClient, 0);
+  });
+
+  it("Defer a message received from recevieDefferedMessage using subscription", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    const sequenceNumber = await deferMsgAndGetSequenceNum(subscriptionClient);
+
+    let deferredMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.defer();
+
+    await testPeekMsgsLength(subscriptionClient, 1);
+
+    if (!sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    deferredMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.complete();
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("Abandon a message received from dead letter queue", async function(): Promise<void> {
+    await queueClient.send(testMessages[0]);
+    const receivedMsgs = await queueClient.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].deadLetter();
+
+    await testPeekMsgsLength(queueClient, 0);
+
+    const deadLetterQueuePath = Namespace.getDeadLetterQueuePathForQueue(queueClient.name);
+    const deadletterQueueClient = namespace.createQueueClient(deadLetterQueuePath);
+    const deadLetterMsgs = await deadletterQueueClient.receiveBatch(1);
+
+    should.equal(deadLetterMsgs.length, 1);
+    should.equal(deadLetterMsgs[0].body, testMessages[0].body);
+    should.equal(deadLetterMsgs[0].messageId, testMessages[0].messageId);
+
+    await deadLetterMsgs[0].abandon();
+
+    const abandonedMsgs = await deadletterQueueClient.receiveBatch(1);
+
+    should.equal(abandonedMsgs.length, 1);
+    should.equal(abandonedMsgs[0].body, testMessages[0].body);
+    should.equal(abandonedMsgs[0].messageId, testMessages[0].messageId);
+
```

Comment:
Since we are abandoning the message, we should check the delivery count

Reply:
I don't know why but abandoning the message received from dead letter queue does not increase the delivery count.

Reply:
That is very interesting, maybe that is intentional. Check with Neeraj if that is expected. Just send a message in the Service Bus channel in Teams